store/cachekv/store.go

```diff
@@ -162,29 +186,32 @@ func (store *Store) Write() {
 }
 
 // CacheWrap implements CacheWrapper.
-func (store *Store) CacheWrap() types.CacheWrap {
-	return NewStore(store)
+func (store *GStore[V]) CacheWrap() types.CacheWrap {
+	return NewGStore(store, store.isZero, store.valueLen)
 }
 
 // CacheWrapWithTrace implements the CacheWrapper interface.
-func (store *Store) CacheWrapWithTrace(w io.Writer, tc types.TraceContext) types.CacheWrap {
-	return NewStore(tracekv.NewStore(store, w, tc))
+func (store *GStore[V]) CacheWrapWithTrace(w io.Writer, tc types.TraceContext) types.CacheWrap {
+	if store, ok := any(store).(*GStore[[]byte]); ok {
+		return NewStore(tracekv.NewStore(store, w, tc))
+	}
+	return store.CacheWrap()
```

Comment:
this part confused me when i looked at it before. is there anyway we can do this so we don't have a special case for the []byte store?

Reply:
Ok, added a comment where we're doing type assertions.